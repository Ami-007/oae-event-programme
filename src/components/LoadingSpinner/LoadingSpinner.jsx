import React from "react";
import "./LoadingSpinner.scss";

const LoadingSpinner = () => {
  return (
    <div className="loading">
      <div className="loading__spinner">
        <div className="loading__bounce1"></div>
        <div className="Loading__loading__bounce2"></div>
        <div className="loading__bounce3"></div>
      </div>
      <h1 className="loading__heading">Loading</h1>
    </div>
  );
};
export default LoadingSpinner;
