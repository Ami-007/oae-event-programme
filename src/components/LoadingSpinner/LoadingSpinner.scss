.loading {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &__heading {
    text-align: center;
  }

  &__spinner {
    width: 4.375rem;
    text-align: center;
  }

  &__spinner > div {
    width: 1.125rem;
    height: 1.125rem;
    background-color: #333;
    border-radius: 100%;
    display: inline-block;
    -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
    animation: sk-bouncedelay 1.4s infinite ease-in-out both;
  }

  &__bounce1 {
    -webkit-animation-delay: -0.32s;
    animation-delay: -0.32s;
  }

  &__bounce2 {
    -webkit-animation-delay: -0.16s;
    animation-delay: -0.16s;
  }

  @-webkit-keyframes sk-bouncedelay {
    0%,
    80%,
    100% {
      -webkit-transform: scale(0);
    }
    40% {
      -webkit-transform: scale(1);
    }
  }

  @keyframes sk-bouncedelay {
    0%,
    80%,
    100% {
      -webkit-transform: scale(0);
      transform: scale(0);
    }
    40% {
      -webkit-transform: scale(1);
      transform: scale(1);
    }
  }
}
