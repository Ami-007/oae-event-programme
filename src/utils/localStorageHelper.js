export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
