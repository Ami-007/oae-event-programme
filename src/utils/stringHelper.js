export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
