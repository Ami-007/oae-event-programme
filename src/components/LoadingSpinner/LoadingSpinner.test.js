import { customRender } from "../../utils/testUtils";
import LoadingSpinner from "./LoadingSpinner";

it("should render the loading spinner", () => {
  const { container } = customRender(<LoadingSpinner />);

  expect(container).toMatchSnapshot();
});
