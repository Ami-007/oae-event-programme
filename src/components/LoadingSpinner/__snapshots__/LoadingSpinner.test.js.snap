// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the loading spinner 1`] = `
<div>
  <div
    class="loading"
  >
    <div
      class="loading__spinner"
    >
      <div
        class="loading__bounce1"
      />
      <div
        class="Loading__loading__bounce2"
      />
      <div
        class="loading__bounce3"
      />
    </div>
    <h1
      class="loading__heading"
    >
      Loading
    </h1>
  </div>
</div>
`;
